### DISCUSSION TITLE
camera lens EXIF model name

### DISCUSSION THREAD

**Discussion by rebio** [lensfun/lensfun#1626] [Q&A] [answered]
From Discussion #1616:
> It is the name that exiv2 returns so that the lens can automatically be found.
I tried get a name with exiv2 with a random image that had. I guess the <model> tag for the database is the content of <Exif.Canon.LensModel> I get with _exifdata_ of exiv2.

The random image was taken with a Canon EF-S 10-22mm lens. The file has the EXIF tag

`<Exif.Canon.LensModel>EF-S10-22mm f/3.5-4.5 USM<Exif.Canon.LensModel/>`
(my guess is that this is the lens name since there is nothing else to find that is clear to read like that)

The lens' <model> tag in the database is

`<model>Canon EF-S 10-22mm f/3.5-4.5 USM</model>`

Is this correct? Does the database entry have to match the EXIF info 1to1 or is enough if there is only a part of it (so eg. "EF-S 10-22mm" in the EXIF would also be enough)?

And what if there is no EXIF lens information? Eg. my smartphone has a maker and a model EXIF tag, but only for the device itself, not for the lens. It has two lenses (one monochrome, the other colour). They do show the same content, but you can access and save their files separately.

I'm asking because of a lens I want to include a new entry and I have no idea which EXIF tag is the right one.

Thank you in advance for the answer! :-)

**Comment by sarunasb** [comment, marked as answer]
I understand you are using something like `exiv2 -pt image.raw |grep -i 'lens\|model'`.
`<model>` element then should contain what is returned by exiv2. You can add info to that line, but you should not remove anything from the string returned by exiv2. Case is not important. Value in `<model>` will be used trying to match the lens. If `<model lang="">` is present, its value can be used by an application to show lens name matching the locale and/or using its more conventional name.

Please check API docs here https://lensfun.github.io/manual/v0.3.2/el_lens.html.

**Reply by rebio** [reply]
Short explanation of the things I did/tested:
I used the _2019msvc64_ build from https://exiv2.org/download.html In the bin folder are a few exe files that I used directly namely exifdata.exe, exifprint.exe and exifvalue.exe

**exifdata.exe:**
Usage: `exifdata.exe path/to/file.jpg xml`
The image contains a lot of data and also the lens information I included as example: `<Exif.Canon.LensModel>EF-S10-22mm f/3.5-4.5 USM<Exif.Canon.LensModel/>`

**exifdata.exe:**
Usage: `exifdata.exe path/to/file.jpg`
The output is formated a bit more human readable and contains more type information, but it's the same value: `Exif.Canon.LensModel                         0x0095 Ascii      64  EF-S10-22mm f/3.5-4.5 USM`

**exifvalue.exe:**
Usage: `exifvalue.exe path/to/file.jpg Exif.Canon.LensModel`
This gives you only the value you requested, but it's not different from the previous outputs: `EF-S10-22mm f/3.5-4.5 USM`

Thanks for the link!
If I read that page right, it actually doesn't really seem to matter that much what the exact model name in the database is because lensfun chops up the name and tries to match it to it's databse.

I was a bit confused that if it has to be a 1to1 match, how is the 10-22mm lens recognised by lensfun if it's database name (`Canon EF-S 10-22mm f/3.5-4.5 USM`) and it's EXIF name `EF-S10-22mm f/3.5-4.5 USM` don't match 1to1. If lensfun does not only make a 1to1 match, it makes much more sense. This also means that I don't have to worry about the <model> tag name in the database that much as long as I copy the lens EXIF info from exiv2.

**Reply by sarunasb** [reply]
Correct.
